src/tree/ContainerAppExtParentTreeItem.ts

```diff
@@ -0,0 +1,39 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import { AzExtParentTreeItem, AzExtTreeItem, IActionContext } from "@microsoft/vscode-azext-utils";
+import { ext } from "../extensionVariables";
+import { ContainerAppsExtResourceBase } from "../resolver/ContainerAppsExtResourceBase";
+
+export class ContainerAppExtParentTreeItem<T extends ContainerAppsExtResourceBase<unknown>> extends AzExtParentTreeItem {
+    public label: string;
+    public contextValue: string;
+    // ideally this would be the branch provider, but that won't work right now
+    treeDataProvider = ext.rgApi.appResourceTree;
+
+    public resource: T;
+
+    constructor(parent: AzExtParentTreeItem | undefined, resource: T) {
+        super(parent);
+        this.label = resource.label;
```

Comment:
This section where we copy all the properties in a constructor gets repeated a lot. We should try to refactor in such a way that uses generics or base classes to deal with these properties just once.

Reply:
I'm having a pretty tough time thinking up a way to refactor this, namely because one needs to extend from `AzExtTreeItem` and the other has to extend from `AzExtParentTreeItem`.

If I refactor them into a base class, I can't have the parent nodes extend that base class because it won't have `AzParentTreeItem`  which leaves me having to implement it.

Maybe just a util function that does all of that initializing so there's less repeated code?

Reply:
It's in a lot more than two places (or at least, it was when I put the comment), right? Just two would be much better.